src/OrbitGl/App.cpp

```diff
@@ -2991,7 +2999,19 @@ OrbitApp::GetConfidenceIntervalEstimator() const {
   return confidence_interval_estimator_;
 }
 
-void OrbitApp::ShowHistogram(const std::vector<uint64_t>* data, const std::string& function_name,
-                             uint64_t function_id) {
-  main_window_->ShowHistogram(data, function_name, function_id);
-}
\ No newline at end of file
+void OrbitApp::ShowHistogram(const std::vector<uint64_t>* data, const std::string& scope_name,
+                             uint64_t scope_id) {
+  main_window_->ShowHistogram(data, scope_name, scope_id);
+}
+
+[[nodiscard]] uint64_t OrbitApp::ProvideId(const orbit_client_protos::TimerInfo& timer_info) const {
+  if (timer_info.function_id() != orbit_grpc_protos::kInvalidFunctionId) {
+    return timer_info.function_id();
+  }
+  if (timer_info.type() == orbit_client_protos::TimerInfo_Type_kApiScope ||
+      timer_info.type() == orbit_client_protos::TimerInfo_Type_kApiScopeAsync) {
+    ORBIT_CHECK(api_event_id_provider_);
+    return api_event_id_provider_->ProvideId(timer_info);
+  }
+  return orbit_grpc_protos::kInvalidFunctionId;
```

Comment:
Discussed offline, mentioning for completeness: you'll need a new UI-only constant for the "scope id" domain, which is different from the "function id" domain.